### ISSUE TITLE
mmcli hangs because of the input file name

### ISSUE THREAD

**Issue by djfm501** [RickStrahl/MarkdownMonster#938]
Here are two tests:

I created two files: `"adding-an-attribute-or-field-to-a-function.md"`
and `a-b.md`.  They contain the same markup text.

I ran two tests:

```
[D:\PyPad\Wiki]mmcli markdowntohtml -i "adding-an-attribute-or-field-to-a-function.md" -o "d:/temp/{660b164b-e225-4b5
6-9b5d-5822843baccd}.html" --theme Dharkan --rendermode packagedhtml
-------------------------------
Markdown Monster Console v2.4.0
-------------------------------
Command: markdowntohtml
^C
[D:\PyPad\Wiki]mmcli markdowntohtml -i "a-b.md" -o "d:/temp/{660b164b-e225-4b56-9b5d-5822843baccd}.html" --theme Dhar
kan --rendermode packagedhtml
-------------------------------
Markdown Monster Console v2.4.0
-------------------------------
Command: markdowntohtml
Created file: d:\temp\{660b164b-e225-4b56-9b5d-5822843baccd}.html
 Output Mode: packagedhtml
```

In the first call to mmcli I had to hit ^c as the program hung.

In the second call mmcli created the desired output html file.

As near as I can tell the only difference is the input file name.

Am I missing something here?

I've attached the input files.  As I said they are both the same.

TIA

David McClelland
[adding-an-attribute-or-field-to-a-function.md](https://github.com/RickStrahl/MarkdownMonster/files/8446299/adding-an-attribute-or-field-to-a-function.md)

**Comment by RickStrahl** [comment]
Isn't this the same as #917?

Can you try latest version please. Pretty sure this was fixed and I verified that it works on my end with test files here.

**Comment by RickStrahl** [comment]
Ah thanks... I ran into this in another of my applications recently and it's due to faulty command line parsing. The issue is the -f or -i in these file names that are being interpreted as commands when they are part of the filename. 

I fixed this in the other app, but not in MM.

**Comment by RickStrahl** [comment]
Fixed two issues:

* Hanging due to invalid file names  
This was caused by attempting to open a file open dialog. This no longer works due to the fact MM now runs an async task operation for all commands, so this was broken as part of the v2 upgrade functionality. Fixed - invalid or missing files now return error messages.

* Invalid Command Line Switch Parsing  
This happen due faulty command line parsing that would incorrectly identify values - rather than switches - as switch values. So values with `-`s could trigger a switch. Fixed. We're now explicitly scanning for argument matches **first** and only if that matches, then parse out the value.

Fixed in 2.4.9.2
